packages/go/analysis/ad/esc10.go

```diff
@@ -0,0 +1,150 @@
+		return nil
+	} else if publishedCertTemplates, ok := cache.PublishedTemplateCache[eca.ID]; !ok {
+		return nil
+	} else {
+		for _, template := range publishedCertTemplates {
+			if valid, err := isCertTemplateValidForESC10a(template); err != nil {
+				if !errors.Is(err, graph.ErrPropertyNotFound) {
+					log.Errorf("Error checking cert template validity for template %d: %v", template.ID, err)
+				} else {
+					log.Debugf("Error checking cert template validity for template %d: %v", template.ID, err)
+				}
+			} else if !valid {
+				continue
+			} else if certTemplateControllers, ok := cache.CertTemplateControllers[template.ID]; !ok {
+				log.Debugf("Failed to retrieve controllers for cert template %d from cache", template.ID)
+				continue
+			} else if ecaControllers, ok := cache.EnterpriseCAEnrollers[eca.ID]; !ok {
+				log.Debugf("Failed to retrieve controllers for enterprise ca %d from cache", eca.ID)
+				continue
+			} else {
+				//Expand controllers for the eca + template completely because we don't do group shortcutting here
+				var (
+					victimBitmap = expandNodeSliceToBitmapWithoutGroups(certTemplateControllers, groupExpansions)
+					ecaBitmap    = expandNodeSliceToBitmapWithoutGroups(ecaControllers, groupExpansions)
+				)
+				victimBitmap.And(ecaBitmap)
+				//Use our id list to filter down to users
+				if userNodes, err := ops.FetchNodeSet(tx.Nodes().Filterf(func() graph.Criteria {
+					return query.And(
+						query.KindIn(query.Node(), ad.User),
+						query.InIDs(query.NodeID(), cardinality.DuplexToGraphIDs(victimBitmap)...),
+					)
+				})); err != nil {
+					if !graph.IsErrNotFound(err) {
+						return err
+					}
+				} else if len(userNodes) > 0 {
+					if subjRequireDns, err := template.Properties.Get(ad.SubjectAltRequireDNS.String()).Bool(); err != nil {
+						log.Debugf("Failed to retrieve subjectAltRequireDNS for template %d: %v", template.ID, err)
+						victimBitmap.Xor(cardinality.NodeSetToDuplex(userNodes))
+					} else if subjRequireDomainDns, err := template.Properties.Get(ad.SubjectAltRequireDomainDNS.String()).Bool(); err != nil {
+						log.Debugf("Failed to retrieve subjectAltRequireDomainDNS for template %d: %v", template.ID, err)
+						victimBitmap.Xor(cardinality.NodeSetToDuplex(userNodes))
+					} else if subjRequireDns || subjRequireDomainDns {
+						//If either of these properties is true, we need to remove all these users from our victims list
+						victimBitmap.Xor(cardinality.NodeSetToDuplex(userNodes))
+					}
+				}
+
+				if attackers, err := ops.FetchStartNodes(tx.Relationships().Filterf(func() graph.Criteria {
+					return query.And(
+						query.KindIn(query.Start(), ad.Group, ad.User, ad.Computer),
+						query.KindIn(query.Relationship(), ad.GenericAll, ad.GenericWrite, ad.Owns, ad.WriteOwner, ad.WriteDACL),
+						query.InIDs(query.EndID(), cardinality.DuplexToGraphIDs(victimBitmap)...),
+					)
+				})); err != nil {
+					return err
```

Comment:
Do we want to exit out of the 10a function here or just continue to the next template in the loop?